### PR TITLE
metrics: add prometheus endpoint

### LINKED ISSUE #671: Expose local metrics via HTTP server
Add simple, optional HTTP server component for serving metrics via. prometheus protocol. HTTP server hopefully flexible so we can add an admin monitoring and health endpoints as well.

### LINKED ISSUE #691: Implement simple HTTP server tile with diagnostics
Little HTTP server. Would serve prometheus metrics, diagnostics, and possibly a cute live view like the monitor command does in a terminal.

### PR DESCRIPTION
**Pull request by mmcgee-jump**
A metrics tile is added and wired up to the metrics shared memory area, with a simple socket based HTTP server that serves the values in Prometheus text format.

Fixes #691 
Fixes #671

### CODE REVIEW

File: src/app/fdctl/run/tiles/fd_metric.c [hunk 1]
```diff
@@ -0,0 +1,476 @@
...
+    fd_topo_tile_t * consumer_tile = &topo->tiles[ i ];
+    for( ulong j=0; j<consumer_tile->in_cnt; j++ ) {
+      if( FD_UNLIKELY( consumer_tile->in_link_id[ j ] == producer->out_link_id_primary && consumer_tile->in_link_reliable[ j ] ) ) {
+        if( FD_UNLIKELY( consumer==consumer_tile && consumer_in_idx==j ) ) return count;
+        count++;
+      }
+    }
+  }
+  return ULONG_MAX;
+}
```
Comment (ptaffet-jump) [inline src/app/fdctl/run/tiles/fd_metric.c:119]: Given all the trouble we had with a similar-looking piece of code yesterday or the day before, this one could use some more comments. I think it finds all reliable consumers of producer's primary output and then gives you the position of consumer within that list, right?

**Review by ptaffet-jump — COMMENTED** [review]
Overall, it's in pretty good shape. A few comments on `fd_metrics.h` mostly

File: src/disco/metrics/generated/fd_metrics_all.h [hunk 2]
```diff
@@ -1,22 +1,145 @@
...
+#define FD_METRICS_HISTOGRAM_STEM_LOOP_FINISH_DURATION_SECONDS_MIN  (5e-08)
+#define FD_METRICS_HISTOGRAM_STEM_LOOP_FINISH_DURATION_SECONDS_MAX  (5e-05)
+#define FD_METRICS_HISTOGRAM_STEM_LOOP_FINISH_DURATION_SECONDS_CVT  (FD_METRICS_CONVERTER_SECONDS)
+
+#define FD_METRICS_HISTOGRAM_STEM_FRAGMENT_FILTERED_SIZE_BYTES_OFF  (139UL)
+#define FD_METRICS_HISTOGRAM_STEM_FRAGMENT_FILTERED_SIZE_BYTES_NAME "stem_fragment_filtered_size_bytes"
+#define FD_METRICS_HISTOGRAM_STEM_FRAGMENT_FILTERED_SIZE_BYTES_TYPE (FD_METRICS_TYPE_HISTOGRAM)
+#define FD_METRICS_HISTOGRAM_STEM_FRAGMENT_FILTERED_SIZE_BYTES_DESC "Size of each fragment that was filtered and not processed by the tile."
+#define FD_METRICS_HISTOGRAM_STEM_FRAGMENT_FILTERED_SIZE_BYTES_MIN  (0UL)
+#define FD_METRICS_HISTOGRAM_STEM_FRAGMENT_FILTERED_SIZE_BYTES_MAX  (2094UL)
```
[the remark this answers is not in the document]
Comment (mmcgee-jump) [reply src/disco/metrics/generated/fd_metrics_all.h:124]: This is `FD_TPU_DCACHE_MTU`

File: src/disco/mux/fd_mux.c [hunk 3]
```diff
@@ -353,19 +360,16 @@ fd_mux_tile( fd_cnc_t *              cnc,
...
-    fd_histf_join( fd_histf_new( hist_filter2_ticks,      0U, max_ticks ) );
-    fd_histf_join( fd_histf_new( hist_filter2_frag_sz,    0U, FD_TPU_DCACHE_MTU ) );
-    fd_histf_join( fd_histf_new( hist_fin_ticks,          0U, max_ticks ) );
-    fd_histf_join( fd_histf_new( hist_fin_frag_sz,        0U, FD_TPU_DCACHE_MTU ) );
+    fd_histf_join( fd_histf_new( hist_housekeeping_ticks, FD_MHIST_SECONDS_MIN( STEM, LOOP_HOUSEKEEPING_DURATION_SECONDS),           FD_MHIST_SECONDS_MAX( STEM, LOOP_HOUSEKEEPING_DURATION_SECONDS ) ) );
+    fd_histf_join( fd_histf_new( hist_backp_ticks,        FD_MHIST_SECONDS_MIN( STEM, LOOP_BACKPRESSURE_DURATION_SECONDS),           FD_MHIST_SECONDS_MAX( STEM, LOOP_BACKPRESSURE_DURATION_SECONDS ) ) );
+    fd_histf_join( fd_histf_new( hist_caught_up_ticks,    FD_MHIST_SECONDS_MIN( STEM, LOOP_CAUGHT_UP_DURATION_SECONDS),              FD_MHIST_SECONDS_MAX( STEM, LOOP_CAUGHT_UP_DURATION_SECONDS ) ) );
+    fd_histf_join( fd_histf_new( hist_ovrnp_ticks,        FD_MHIST_SECONDS_MIN( STEM, LOOP_OVERRUN_POLLING_DURATION_SECONDS),        FD_MHIST_SECONDS_MAX( STEM, LOOP_OVERRUN_POLLING_DURATION_SECONDS ) ) );
```
[the remark this answers is not in the document]
Comment (mmcgee-jump) [reply src/disco/mux/fd_mux.c:366]: Mildly opposed for now, the metrics use should be rare and verbose. We probably can add a macro for this histogram creation stuff but need more consumers to see what a good pattern would be first.

**Review by ptaffet-jump — APPROVED** [review]

**Pull request merged**